rts/Map/SMF/ROAM/Patch.cpp

```diff
@@ -228,8 +240,12 @@ bool Patch::Split(TriTreeNode* tri)
 		return true;
 
 	// if this triangle is not in a proper diamond, force split our base-neighbor
-	if (!tri->BaseNeighbor->IsDummy() && (tri->BaseNeighbor->BaseNeighbor != tri))
+	if (!tri->BaseNeighbor->IsDummy() && (tri->BaseNeighbor->BaseNeighbor != tri)){
 		Split(tri->BaseNeighbor);
+        if (tri->BaseNeighbor->parentPatch != this){
```

Comment:
for consistency with rest of code, don't use  {} when there's a single statement